cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -155,91 +148,99 @@ def get_splice_junctions_bed(self, case_id: str, sample_id: str) -> Optional[hk_
-        `scout load gene-fusion-report [-r] <case_id> <path/to/research_gene_fusion_report.pdf>`
-
-        Args:
-            dry_run     (bool):         Skip uploading
-            case_id     (string):       Case identifier
-            research    (bool):         Upload research report instead of clinical
-            update      (bool):         Overwrite existing report
-        Returns:
-            Nothing
-        """
+        """Upload fusion report file for a case to Scout."""
 
-        scout_api: ScoutAPI = self.scout
-        status_db: Store = self.status_db
         report_type: str = "Research" if research else "Clinical"
-        rna_case: models.Family = status_db.family(case_id)
 
-        rna_dna_sample_case_map: Dict[str, Dict[str, list]] = self.create_rna_dna_sample_case_map(
-            rna_case=rna_case
-        )
-        unique_dna_cases: Set[str] = set()
         fusion_report: Optional[hk_models.File] = self.get_fusion_report(case_id, research)
         if fusion_report is None:
             raise FileNotFoundError(
                 f"{report_type} fusion report was not found in housekeeper for {case_id}"
             )
 
         LOG.info(f"{report_type} fusion report {fusion_report.path} found")
-        for rna_sample_id in rna_dna_sample_case_map:
-            dna_cases: List[str]
-            dna_sample_id, dna_cases = rna_dna_sample_case_map[rna_sample_id].popitem()
-            unique_dna_cases.update(dna_cases)
 
-        for dna_case_id in unique_dna_cases:
+        for dna_case_id in self.get_unique_dna_cases_related_to_rna_case(case_id):
             LOG.info(f"Uploading {report_type} fusion report to scout for case {dna_case_id}")
 
             if dry_run:
                 continue
-            scout_api.upload_fusion_report(
+            self.scout_api.upload_fusion_report(
                 case_id=dna_case_id,
                 report_path=fusion_report.full_path,
                 research=research,
                 update=update,
             )
-            LOG.info("Uploaded %s fusion report", report_type)
+            LOG.info(
+                f"Uploaded {report_type} fusion report",
+            )
 
-        LOG.info("Upload %s fusion report finished!", report_type)
+        LOG.info(f"Upload {report_type} fusion report finished!")
+
+    def upload_rna_report_to_scout(
+        self,
+        dry_run: bool,
+        report_type: str,
+        report_file: hk_models.File,
+        rna_case_id: str,
+    ) -> None:
+        """Upload report file to DNA cases related to a RNA case in scout."""
+        LOG.info(f"Finding DNA cases related to RNA case {rna_case_id}")
+        for dna_case_id in self.get_unique_dna_cases_related_to_rna_case(rna_case_id):
+            self.upload_report_to_scout(
+                dry_run=dry_run,
+                report_type=report_type,
+                report_file=report_file,
+                case_id=dna_case_id,
+            )
+
+    def upload_report_to_scout(
```

Comment:
oh I see you have this function here. Why not using it within the other upload functions for specific reports?